Tutorials/Creating-a-Property-Editor/part-2-v9.md

```diff
@@ -0,0 +1,140 @@
+
+First create a `SuggestionConfiguration.cs` file:
+
+```csharp
+namespace Umbraco.Cms.Core.PropertyEditors
+{
+    public class SuggestionConfiguration
+    {
+        [ConfigurationField("isEnabled", "Enabled?", "boolean", Description = "Provides Suggestions")]
+        public bool Enabled { get; set; }
+    }
+}
+```
+
+Then create a `SuggestionConfigurationEditor.cs` file:
+
+```csharp
+using Umbraco.Cms.Core.IO;
+
+namespace Umbraco.Cms.Core.PropertyEditors
+{
+     public class SuggestionConfigurationEditor : ConfigurationEditor<SuggestionConfiguration>
+    {
+        public SuggestionConfigurationEditor(IIOHelper ioHelper) : base(ioHelper)
+        {
+        }
+    }
+
+}
+```
+
+Finally, edit the `Suggestion.cs` file from step one until it looks like the example below:
+
+```csharp
+using Umbraco.Cms.Core.IO;
+
+namespace Umbraco.Cms.Core.PropertyEditors
+{
+    [DataEditor(
+        alias: "Suggestions editor",
+        name: "Suggestions Editor",
+        view: "~/App_Plugins/Suggestions/suggestion.html",
+        Group = "Common",
+        Icon = "icon-list")]
+    public class Suggestions : DataEditor
+    {
+        private readonly IIOHelper _ioHelper;
+        public Suggestions(IDataValueEditorFactory dataValueEditorFactory,
+            IIOHelper ioHelper)
+            : base(dataValueEditorFactory)
+
+        {
+            _ioHelper = ioHelper;
+        }
+        protected override IConfigurationEditor CreateConfigurationEditor() => new SuggestionConfigurationEditor(_ioHelper);
+        
+    }
+}
+```
+
+So what did we add? We added a prevalue editor, with a `fields` collection. This collection contains information about the UI we will render on the data type configuration for this editor.
+
+The label "Enabled?" uses the view "boolean", so this will allow us to turn the suggestions on/off and will provide the user with a toggle button. The name "boolean" comes from the convention that all preview editors are stored in `/umbraco/views/prevalueeditors/` and then found via `boolean.html`
+
+Save the manifest, restart the application and have a look at the markdown data type in Umbraco now. You should now see that you have 1 configuration option:
+
+![An example of how the configuration will look](images/suggestion-editor-config.png)
+
+## Using the configuration
+
+The next step is to gain access to our new configuration options. For this, open the `suggestion.controller.js` file.
+
+Let's first add the default value functionality. When the `$scope.model.value` is empty or *undefined*, we want to use the default value. To do that, we add the following to the very beginning of the controller:
+
+```javascript
+if($scope.model.value === null || $scope.model.value === ""){
+    $scope.model.value = $scope.model.config.defaultValue;
+}
+```
+
+and the nat the end we add a getState method:
```

Comment:
Is the "nat" on purpose? 

Not sure I completely understand the sentence 😅

Reply:
Oops, Good catch. That line is supposed to be "and then at the end.." just a space typo..